c_glib/arrow-flight-glib/client.cpp

```diff
@@ -316,8 +316,8 @@ gaflight_client_list_flights(GAFlightClient *client,
   }
   std::unique_ptr<arrow::flight::FlightListing> flight_listing;
   auto status = flight_client->ListFlights(*flight_options,
-                                           *flight_criteria,
-                                           &flight_listing);
+                                           *flight_criteria
+                               ).Value(&flight_listing);
```

Comment:
Similarly, this (and below) would benefit from splitting the result and the status into separate statements so that we can keep a consistent style.